test/filter.test.ts

```diff
@@ -83,13 +83,31 @@ describe("FilterCheck", () => {
 			Optional(Number.withConstraint((n) => n > 5))
 		);
 
-		expect(checkWrapper(10)).toEqual(10);
+		expect(checkWrapper(10)).toBe(10);
+		expect(checkWrapper(undefined)).toBeUndefined();
 
 		const checkChain = FilterCheck(
 			Number.optional().withConstraint((n) => typeof n === "undefined" || n > 5)
 		);
 
-		expect(checkChain(10)).toEqual(10);
+		expect(checkChain(10)).toBe(10);
+		expect(checkChain(undefined)).toBeUndefined();
+
+		const checkRecord = FilterCheck(
+			Record({
+				a: Number.withConstraint((n) => n > 5).optional(),
+			}).optional()
+		);
+
+		expect(checkRecord({ a: 10 })).toStrictEqual({ a: 10 });
+		expect(checkRecord({})).toStrictEqual({});
+		expect(checkRecord(undefined)).toBeUndefined();
```

Comment:
On `main` this would throw:

```typescript
TypeError: Cannot convert undefined or null to object
    at hasOwnProperty (<anonymous>)

  69 |              case "record":
  70 |                      return Object.keys(r.fields)
> 71 |                              .filter((k) => Object.prototype.hasOwnProperty.call(x, k))
     |                                                                             ^
  72 |                              .reduce(
  73 |                                      (acc, k) => ({
  74 |                                              ...acc,

  at src/index.ts:71:52
      at Array.filter (<anonymous>)
  at filterInternal (src/index.ts:71:6)
  at filterInternal (src/index.ts:85:11)
  at filter (src/index.ts:96:9)
  at src/index.ts:158:16
  at Object.<anonymous> (test/filter.test.ts:104:10)
```